core/arch/aarch64/codec.txt

```diff
@@ -271,14 +271,14 @@ x0110111xxxxxxxxxxxxxxxxxxxxxxxx  tbnz   tbz
 00001000010xxxxx0xxxxxxxxxxxxxxx  ldxrb  w0 : mem0 ign10 ign16
 00001000010xxxxx1xxxxxxxxxxxxxxx  ldaxrb w0 : mem0 ign10 ign16
 00001000100xxxxx1xxxxxxxxxxxxxxx  stlrb  mem0 : w0 ign10 ign16
-00001000110xxxxx1xxxxxxxxxxxxxxx  ldarb  w0 : mem0 ign10 ign16
+0000100011011111111111xxxxxxxxxx  ldarb  w0 : x5sp
```

Comment:
I am not sure if this is the right thing to do for LDARB (and the related instructions) This instruction should take a memory operand, not a register operand. Even though for those instruction, only memory references without offset are allowed, we should not hide the fact that we access memory here. Unless I am mistaken, memory operands are used to check if the instruction accesses memory and making this a register operand could cause problems.

Ideally, ldarb should only allow memory operands with byte size, ldarh 2bytes and so on

Reply:
Yes, if the register is a base register used to refer to memory, it should be a memory operand, not a register operand.

Reply:
Good call! This passed me by :-/
Using memory operand results in encoding failure. Looking at it now . . .